File: src/ts/sort/insertionsort.ts
// InsertionSort sorts array of items
export function insertionSort(arr: number[]) {
  for (let i = 1; i < arr.length; i++) {
    const key = arr[i]
    let j = i - 1
    for (; j >= 0 && key < arr[j]; j--) {
      arr[j+1] = arr[j]
    }
    arr[j+1] = key
  }
}

File: go/src/sort/insertionsort.go
package sort

// InsertionSort sorts array of items
func InsertionSort(arr []Item) {
	n := len(arr)
	for i := 1; i < n; i++ {
		key := arr[i]
		j := i - 1
		for ; j >= 0 && key.Less(arr[j]); j-- {
			arr[j+1] = arr[j]
		}
		arr[j+1] = key
	}
}
